packages/server/lib/routes.ts

```diff
@@ -199,6 +199,7 @@ web.route('/api/v1/integration/:providerConfigKey/connections').get(webAuth, con
 web.route('/api/v1/provider').get(configController.listProvidersFromYaml.bind(configController));
 
 web.route('/api/v1/connection').get(webAuth, connectionController.listConnections.bind(connectionController));
+web.route('/api/v1/connection/metadata').patch(webAuth, updateMetadata);
```

Comment:
I definitely think you can create a dedicated endpoint, i.e: `PUT api/v1/connection/config`